client/db/bolt/upgrades.go

```diff
@@ -190,21 +206,39 @@ func ensureVersion(tx *bbolt.Tx, ver uint32) error {
 	return nil
 }
 
-func reloadMatchProofs(tx *bbolt.Tx) error {
+// Note that reloadMatchProofs will rewrite the MatchProof with the current
+// match proof encoding version. Thus, multiple upgrades in a row calling
+// reloadMatchProofs may be no-ops. Matches with cancel orders may be skipped.
+func reloadMatchProofs(tx *bbolt.Tx, skipCancels bool) error {
 	matches := tx.Bucket(matchesBucket)
 	return matches.ForEach(func(k, _ []byte) error {
 		mBkt := matches.Bucket(k)
 		if mBkt == nil {
 			return fmt.Errorf("match %x bucket is not a bucket", k)
 		}
-		proofB := getCopy(mBkt, proofKey)
+		proofB := mBkt.Get(proofKey)
 		if len(proofB) == 0 {
 			return fmt.Errorf("empty match proof")
 		}
-		proof, err := dexdb.DecodeMatchProof(proofB)
+		proof, ver, err := dexdb.DecodeMatchProof(proofB)
 		if err != nil {
 			return fmt.Errorf("error decoding proof: %w", err)
 		}
+		// No need to rewrite this if it was loaded from the current version.
+		if ver == dexdb.MatchProofVer {
+			return nil
+		}
```

Comment:
This logic would make resuming from failed batches easily doable, for this upgrade, if we did batches.